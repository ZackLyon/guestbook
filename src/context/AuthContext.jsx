import React, { createContext, useContext, useState } from 'react';

const AuthContext = createContext();

const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  const login = ({ username, password }) => {
    const didLoginSucceed =
      username === process.env.USERNAME_AUTH &&
      password === process.env.PASSWORD_AUTH;

    if (didLoginSucceed) setUser(username);

    return didLoginSucceed;
  };

  const logout = (loggingOut) => {
    setUser(null);
    loggingOut();
  };

  return (
    <AuthContext.Provider value={{ user, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

const useAuth = () => {
  const context = useContext(AuthContext);

  if (context === undefined)
    throw new Error(
      'useAuth needs to be used on a descendant element of the Provider'
    );

  return context;
};

export { AuthProvider, useAuth };
