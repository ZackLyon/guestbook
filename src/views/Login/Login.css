.loginPage {
  background-color: black;
  height: 90vh;
  width: 100vw;
  display: flex;
  justify-content: center;
}

.loginForm {
  position: relative;
  top: 5vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 10vmax;
  width: 20vw;
  padding: 1vh;
  border-radius: 0.5vh;
  background-color: white;
  box-shadow: 0.2vmax 0.2vmax 0 0.2vmax salmon;
}

label {
  font-weight: bold;
}

.loginForm > input {
  outline-color: salmon;
  font-size: 1.2vmax;
  margin-bottom: 1vh;
}

.loginForm > button {
  color: white;
  background-color: seagreen;
  box-shadow: 0.1vmax 0.1vmax 0 0.05vmax salmon;
  padding: 0.5vmax;
  border: none;
  border-radius: 0.5vh;
  font-weight: bold;
  cursor: pointer;
}
