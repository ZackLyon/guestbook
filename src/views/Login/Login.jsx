import React, { useState } from 'react';
import { useAuth } from '../../context/AuthContext.jsx';
import { useHistory } from 'react-router-dom';
import './Login.css';

export default function Login() {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState(null);
  const auth = useAuth();
  const history = useHistory();
  const { login } = auth;

  const handleSubmit = (e) => {
    e.preventDefault();
    const successfulLogin = login({ username, password });
    console.log(successfulLogin);
    console.log(username, password);

    successfulLogin
      ? history.replace('/')
      : setError('Incorrect credentials. Please log in.');
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label htmlFor="username">Username</label>
        <input
          id="username"
          name="username"
          value={username}
          onChange={({ target }) => setUsername(target.value)}
        ></input>
        <label htmlFor="password">Password</label>
        <input
          id="password"
          name="password"
          value={password}
          onChange={({ target }) => setPassword(target.value)}
        ></input>
        <button type="submit">Submit</button>
      </form>
      {error && <div>{error}</div>}
    </div>
  );
}
