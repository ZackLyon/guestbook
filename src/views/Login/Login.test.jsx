import { render } from '@testing-library/react';
import Login from './Login';
import { AuthProvider } from '../../context/AuthContext.jsx';

it('should display Login view on the screen', () => {
  const { container } = render(
    <AuthProvider>
      <Login />
    </AuthProvider>
  );

  expect(container).toMatchSnapshot();
});
