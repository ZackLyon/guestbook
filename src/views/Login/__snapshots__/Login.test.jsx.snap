// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display Login view on the screen 1`] = `
<div>
  <div
    class="loginPage"
  >
    <form
      class="loginForm"
    >
      <label
        for="username"
      >
        Username
      </label>
      <input
        id="username"
        name="username"
        value=""
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
